src/esp/sensor/VisualSensor.h

```diff
@@ -81,14 +90,51 @@ class VisualSensor : public Sensor {
     return false;
   }
 
+  /**
+   * @brief Sets resolution of Sensor's sensorSpec
+   */
+  void setResolution(int height, int width);
+  void setResolution(vec2i resolution);
+
   /**
    * @brief Returns RenderCamera
    */
   virtual gfx::RenderCamera* getRenderCamera() = 0;
 
+  /**
+   * @brief Gets near plane distance.
+   */
+  float getNear() { return near_; }
+
+  /**
+   * @brief Gets far plane distance.
+   */
+  float getFar() { return far_; }
+
+  /**
+   * @brief Returns the FOV of this Sensor
+   */
+  Mn::Deg getFOV() const { return hfov; }
+
  protected:
-  std::unique_ptr<gfx::RenderTarget> tgt_;
+  /** @brief projection parameters
+   */
+
+  /** @brief near clipping plane
+   */
+  float near_ = 0.001f;
 
+  /** @brief far clipping plane
+   */
+  float far_ = 1000.0f;
+
+  /** @brief field of view
+   */
+  Mn::Deg hfov = Mn::Deg{90.f};
```

Comment:
you do not have to have this variable. It is redundant, and you need to maintain it every time when width or height is updated, which is rather error-prone.

in the `getFOV()` just compute it every time.